### PR TITLE
Contact event routes

### PR DESCRIPTION
**Pull request by xunchuan-liu**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -116,12 +118,37 @@ mod tests {
                     asset_tag.try_into().expect("Failed to create initial asset tag")
                 };
```
Comment (xunchuan-liu) [inline src/main.rs:120]: Added initial location in database for tests

File: src/main.rs [hunk 2]
```diff
@@ -116,12 +118,37 @@ mod tests {
...
+                    let location = locations::Location::create(locations::MaybeLocation {
+                        name: Some(String::from("initial")),
+                        latitude: 1.0,
+                        longitude: 1.0,
+                        ip: Some(IpNetwork::V4("10.9.0.32/16".parse().unwrap())),
+                    })
+                    .expect("Failed to create test location");
+                    location.try_into().expect("Failed to create initial location")
+                };
+
```
Comment (xunchuan-liu) [inline src/main.rs:132]: Added initial alert in database for testing

File: src/main.rs [hunk 3]
```diff
@@ -116,12 +118,37 @@ mod tests {
...
 
     pub fn setup() {
         lazy_static::initialize(&FIXTURE);
         lazy_static::initialize(&ADMIN_USER);
         lazy_static::initialize(&INITIAL_ASSET_TAG);
+        lazy_static::initialize(&INITIAL_LOCATION);
+        lazy_static::initialize(&INITIAL_ALERT);
     }
 
     #[derive(Serialize, Deserialize)]
```
Comment (xunchuan-liu) [inline src/main.rs:154]: Modified alert tests to account for initial value

File: src/main.rs [hunk 4]
```diff
@@ -855,14 +882,14 @@ mod tests {
             )
             .to_request();
         let resp: Vec<alerts::Alert> = test::read_response_json(&mut app, req).await;
-        assert_eq!(resp.len(), 0);
+        assert_eq!(resp.len(), 1);
     }
 
     #[actix_rt::test]
     async fn test_location_resource() {
         setup();
```
Comment (xunchuan-liu) [inline src/main.rs:891]: Modified location tests to account for initial value

**Review by trueb2 — DISMISSED** [review]

**Review by trueb2 — APPROVED** [review]

**Pull request merged**